File: translations/codenet/deepseek_commented/C++2P/gpt-4o/Code/s495709932.py
def main():
    import sys
    input = sys.stdin.read
    data = input().split()
    
    N = int(data[0])
    A = list(map(int, data[1:N+1]))

    answer = 0
    # Calculate the maximum absolute difference between any two elements in the array
    for i in range(N - 1):
        for j in range(i + 1, N):
            diff = abs(A[i] - A[j])
            answer = max(answer, diff)
    
    # Output the result
    print(answer)

if __name__ == "__main__":
    main()

# <END-OF-CODE>

File: translations/codenet/deepseek_commented/C++2J/gpt-4o/Code/s495709932.java
import java.util.Scanner;

public class Main {
    public static void main(String[] args) {
        Scanner scanner = new Scanner(System.in);
        int N = scanner.nextInt();
        long[] A = new long[N];
        
        for (int i = 0; i < N; i++) {
            A[i] = scanner.nextLong();
        }

        long answer = 0;
        // Calculate the maximum absolute difference between any two elements in the array
        for (int i = 0; i < N - 1; i++) {
            for (int j = i + 1; j < N; j++) {
                long diff = Math.abs(A[i] - A[j]);
                answer = Math.max(answer, diff);
            }
        }
        // Output the result
        System.out.println(answer);
        scanner.close();
    }
}
// <END-OF-CODE>
